integration-tests/tests/src/tests/objects.ts

```diff
@@ -17,244 +17,1320 @@
 ////////////////////////////////////////////////////////////////////////////
 import { expect } from "chai";
 import Realm from "realm";
-
+import { Buffer } from "buffer";
 import { IPerson, Person, PersonSchema } from "../schemas/person-and-dogs";
 import {
   IPerson as IPersonWithId,
   Person as PersonWithId,
   PersonSchema as PersonSchemaWithId,
 } from "../schemas/person-and-dog-with-object-ids";
+import { openRealmBeforeEach } from "../hooks";
+import { expectArraysEqual, expectSimilar } from "../utils/comparisons";
```

Comment:
In a future refactor, we could consider finding ways to avoid relying on these special comparisons.